File: fixtures/scale/python/large/cipherlab/checks/check_zigzagpack.py
from cipherlab.ciphers.zigzagpack import zigzagpack_encode, zigzagpack_decode
from cipherlab.specs.zigzagpack_spec import zigzagpack_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_zigzagpack():
    spec = zigzagpack_spec()
    for text in corpus_codec():
        packed = zigzagpack_encode(text)
        if zigzagpack_decode(packed) != text:
            return False
    return spec["category"] == "codec"

File: fixtures/scale/ruby/large/lib/check_zigzagpack.rb
require_relative 'zigzagpack_encode'
require_relative 'zigzagpack_decode'
require_relative 'corpus_codec'

def check_zigzagpack
  corpus_codec.each do |text|
    packed = zigzagpack_encode(text)
    return false if zigzagpack_decode(packed) != text
  end
  true
end
